### PR TITLE
docs: document dictionary responses

### LINKED ISSUE #370: document collections response types
Document collections response types and remove nowarn directrive from csproj file

- [x] dictionary
- [x] set
- [x] list

### PR DESCRIPTION
**Pull request by malandis**
Document set, dictionary, and list responses. Enables compiler warning for undocumented publicly visible things; zero warnings after this PR.

Closes #370

### CODE REVIEW

**Review by cprice404 — APPROVED** [review]
hero

File: src/Momento.Sdk/Responses/CacheDictionaryFetchResponse.cs [hunk 1]
```diff
@@ -9,15 +9,48 @@
...
+/// {
+///     return hitResponse.ValueDictionaryStringString;
+/// }
+/// else if (response is CacheDictionaryFetchResponse.Error errorResponse)
+/// {
+///     // handle error as appropriate
+/// }
+/// else
+/// {
+///     // handle unexpected response
```
Comment (nand4011) [inline src/Momento.Sdk/Responses/CacheDictionaryFetchResponse.cs:34]: Why does the example lump miss into the unexpected responses?

File: src/Momento.Sdk/Responses/CacheDictionaryGetFieldResponse.cs [hunk 2]
```diff
@@ -1,44 +1,95 @@
...
+/// {
+///     return hitResponse.ValueString;
+/// }
+/// else if (response is CacheDictionaryGetFieldResponse.Error errorResponse)
+/// {
+///     // handle error as appropriate
+/// }
+/// else
+/// {
+///     // handle unexpected response
```
Comment (nand4011) [inline src/Momento.Sdk/Responses/CacheDictionaryGetFieldResponse.cs:33]: Same here.

File: src/Momento.Sdk/Responses/CacheDictionaryGetFieldsResponse.cs [hunk 3]
```diff
@@ -11,15 +11,53 @@
...
+/// {
+///     return hitResponse.ValueDictionaryStringString;
+/// }
+/// else if (response is CacheDictionaryGetFieldsResponse.Error errorResponse)
+/// {
+///     // handle error as appropriate
+/// }
+/// else
+/// {
+///     // handle unexpected response
```
Comment (nand4011) [inline src/Momento.Sdk/Responses/CacheDictionaryGetFieldsResponse.cs:36]: Same here.

File: src/Momento.Sdk/Responses/CacheListFetchResponse.cs [hunk 4]
```diff
@@ -9,14 +9,47 @@
...
+/// {
+///     return hitResponse.ValueListString;
+/// }
+/// else if (response is CacheListFetchResponse.Error errorResponse)
+/// {
+///     // handle error as appropriate
+/// }
+/// else
+/// {
+///     // handle unexpected response
```
Comment (nand4011) [inline src/Momento.Sdk/Responses/CacheListFetchResponse.cs:34]: Same here. I'll stop commenting on these individually.

**Comment by malandis** [comment]
Will get to the hit/miss/error issues in a follow up PR

**Pull request merged**